sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventParticipantFacadeEjb.java

```diff
@@ -349,6 +351,79 @@ public EventParticipantDto saveEventParticipant(@Valid EventParticipantDto dto,
 		return toPseudonymizedDto(entity, pseudonymizer);
 	}
 
+	/**
+	 * returns a list containing event participants of the given two persons in the same event
+	 * 
+	 * @param firstPersonUuid
+	 * @param secondPersonUuid
+	 * @return
+	 */
+	public List<EventParticipantSelectionDto> getEventParticipantsWithSameEvent(String firstPersonUuid, String secondPersonUuid) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<EventParticipantSelectionDto> cq = cb.createQuery(EventParticipantSelectionDto.class);
+		Root<Event> eventRoot = cq.from(Event.class);
+		Join<Event, EventParticipant> eventParticipantJoin = eventRoot.join(Event.EVENT_PERSONS, JoinType.INNER);
+		Join<EventParticipant, Person> personJoin = eventParticipantJoin.join(EventParticipant.PERSON, JoinType.INNER);
+		Join<EventParticipant, Case> caseJoin = eventParticipantJoin.join(EventParticipant.RESULTING_CASE, JoinType.LEFT);
+		Join<EventParticipant, District> districtJoin = eventParticipantJoin.join(EventParticipant.DISTRICT, JoinType.LEFT);
+
+		cq.multiselect(
+			eventRoot.get(Event.UUID),
+			eventRoot.get(Event.EVENT_TITLE),
+			eventParticipantJoin.get(EventParticipantDto.UUID),
+			personJoin.get(PersonDto.UUID),
+			personJoin.get(PersonDto.FIRST_NAME),
+			personJoin.get(PersonDto.LAST_NAME),
+			personJoin.get(PersonDto.APPROXIMATE_AGE),
+			personJoin.get(PersonDto.APPROXIMATE_AGE_TYPE),
+			personJoin.get(PersonDto.BIRTH_DATE_DD),
+			personJoin.get(PersonDto.BIRTH_DATE_MM),
+			personJoin.get(PersonDto.BIRTH_DATE_YYYY),
+			personJoin.get(PersonDto.SEX),
+			districtJoin.get(District.NAME),
+			eventParticipantJoin.get(EventParticipantDto.INVOLVEMENT_DESCRIPTION),
+			caseJoin.get(Case.UUID));
+
+		cq.where(
+			cb.and(
+				createPersonEventSubquery(cq, cb, firstPersonUuid, eventRoot),
+				createPersonEventSubquery(cq, cb, secondPersonUuid, eventRoot),
+				cb.in(personJoin.get(Person.UUID)).value(Arrays.asList(firstPersonUuid, secondPersonUuid))));
+
+		List<EventParticipantSelectionDto> resultList = em.createQuery(cq).getResultList();
+
+		if (!resultList.isEmpty()) {
+			EventParticipantCriteria eventParticipantCriteria = new EventParticipantCriteria();
+			Map<String, Long> eventParticipantContactCount = getContactCountPerEventParticipant(
+				resultList.stream().map(EventParticipantSelectionDto::getUuid).collect(Collectors.toList()),
+				eventParticipantCriteria);
+
+			for (EventParticipantSelectionDto eventParticipantSelectionDto : resultList) {
+				Optional.ofNullable(eventParticipantContactCount.get(eventParticipantSelectionDto.getUuid()))
+					.ifPresent(eventParticipantSelectionDto::setContactCount);
+			}
+		}
+		return resultList;
+	}
+
+	private Predicate createPersonEventSubquery(
```

Comment:
Please change the signature to:
```
isPersonParticipantInEvent(
                String personUuid
                Root<Event> event
		CriteriaQuery<EventParticipantSelectionDto> cq,
		CriteriaBuilder cb)
```

Then the above query becomes
```
cq.where(
  cb.and(
    isPersonParticipantInEvent(firstPersonUuid, eventRoot, cq, cb),
    isPersonParticipantInEvent(secondPersonUuid, eventRoot, cq, cb),
   cb.in(personJoin.get(Person.UUID)).value(Arrays.asList(firstPersonUuid, secondPersonUuid))
));
```
This way it is easier what those queries do

Reply:
signature updated